app/faucet/views.py

```diff
@@ -0,0 +1,126 @@
+
+def faucet(request):
+    faucet_amount = getattr(settings, "FAUCET_AMOUNT", .003)
+    params = {
+        'title': 'Faucet',
+        'faucet_amount': faucet_amount
+    }
+
+    return TemplateResponse(request, 'faucet_form.html', params)
+
+def github_profile(request, profile):
+    if FaucetRequest.objects.user(profile):
+      response = {
+          'status': 200,
+          'name': profile
+      }
+      return JsonResponse(response)
+    else:
+      response = check_github(profile)
+      return JsonResponse(response)
+
+def check_github(profile):
+  if settings.ENV == 'local':
+    creds = False
+  else:
+    creds = True
+  user = search(profile + ' in:login type:user', creds)
+  if len(user['items']) == 0 or user['items'][0]['login'].lower() != profile.lower():
+    response = {
+      'status': 200,
+      'user': False
+    }
+  else:
+    response = {
+      'status': 200,
+      'user': user['items'][0]
+    }
+  return response
+
+def save_faucet(request):
+    try:
+        validate_slug(request.POST.get('githubProfile'))
+    except Exception as e:
+        return JsonResponse({
+          'message': e.messages[0]
+        }, status=400)
+
+    try:
+        validate_email(request.POST.get('emailAddress'))
+    except Exception as e:
+        return JsonResponse({
+          'message': e.messages[0]
+        }, status=400)
+
+    try:
+        validate_slug(request.POST.get('ethAddress'))
+    except Exception as e:
+        return JsonResponse({
+          'message': e.messages[0]
+        }, status=400)
+    githubProfile = request.POST.get('githubProfile')
+    emailAddress = request.POST.get('emailAddress')
+    ethAddress = request.POST.get('ethAddress')
+    comment = escape(strip_tags(request.POST.get('comment')))
+    checkeduser = check_github(githubProfile)
+    print checkeduser
+    if FaucetRequest.objects.user(githubProfile):
+        return JsonResponse({
+            'message': 'The submitted github profile shows a previous faucet distribution.'
+        }, status=403)
+    elif checkeduser == False:
+      return JsonResponse({
+        'message': 'The submitted github profile could not be found on github.'
+      }, status=400)
+    else:
+      githubMeta = checkeduser
+
+    fr = FaucetRequest(
+        fulfilled=False,
+        github_username = githubProfile,
+        github_meta = githubMeta,
+        address = ethAddress,
+        email = emailAddress,
+        comment = comment
+    )
+
+    fr.save()
+    from_email = settings.PERSONAL_CONTACT_EMAIL
+    to_email = settings.SERVER_EMAIL
+    subject = "New Faucet Request"
+    body = "A new faucet request was completed. You may fund the request here : https://gitcoin.co/_administration/process_faucet_request/[pk]"
+    send_mail(from_email, to_email, subject, body.replace('[pk]', fr.pk), from_name="No Reply from Gitcoin.co")
+    return JsonResponse({
+      'message': 'Created.'
+    }, status=201)
+
+def process_faucet_request(request, pk):
```

Comment:
should this perhaps be an admin-only view?

Reply:
yes... yes it should.... I will check how to set that.